modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/consensus/ConsensusManager.scala

```diff
@@ -1,73 +1,103 @@
 package org.tessellation.sdk.infrastructure.consensus
 
-import cats.effect.{Async, Spawn}
+import cats.effect._
 import cats.kernel.Next
 import cats.syntax.applicativeError._
 import cats.syntax.flatMap._
 import cats.syntax.functor._
+import cats.syntax.option._
 import cats.syntax.show._
 import cats.syntax.traverse._
 import cats.{Applicative, Order, Show}
 
+import scala.concurrent.duration.FiniteDuration
 import scala.reflect.ClassTag
 import scala.reflect.runtime.universe.TypeTag
 
 import org.tessellation.ext.cats.syntax.next._
-import org.tessellation.sdk.domain.consensus.ConsensusFunctions
-import org.tessellation.security.signature.Signed
+import org.tessellation.sdk.infrastructure.consensus.trigger.{ConsensusTrigger, EventTrigger, TimeTrigger}
 
 import org.typelevel.log4cats.slf4j.Slf4jLogger
 
-trait ConsensusManager[F[_], Event, Key, Artifact] {
+trait ConsensusManager[F[_], Key, Artifact] {
 
-  def checkForTrigger(event: Event): F[Unit]
-  def checkForStateUpdate(key: Key)(resources: ConsensusResources[Artifact]): F[Unit]
-  def checkForStateUpdateSync(key: Key)(resources: ConsensusResources[Artifact]): F[Unit]
+  /** Schedules facilitating consensus with `TimeTrigger` after configured interval */
+  def scheduleTimeTrigger: F[Unit]
+
+  /** Tries to facilitate consensus immediately without any `TimeTrigger` or EventTrigger` */
+  def triggerWithNothing: F[Unit]
+
+  private[consensus] def triggerWithEvent: F[Unit]
```

Comment:
Maybe we should use different names to avoid comments. Like:

`trigger`, `triggerWithEvent`, `triggerOnSchedule`

Reply:
Ok, I can remove the comments, but I would keep `triggerWithNothing` to be explicit about it.

Reply:
In the end, I replaced `triggerWithNothing` with `triggerOnStart` as it's used only initially.